### PR TITLE
[IBCDPE-149] Implement new SAMPLE_ID validation

### LINKED ISSUE #443: Add validation check for SAMPLE_ID
cBioPortal file imports have an undocumented character limit of 50 for SAMPLE_IDs.  This caused issues with the 12.2-consortium release with one site's IDs.

### PR DESCRIPTION
**Pull request by thomasyu888**
Fixes #443 
* SAMPLE_ID must start with GENIE-CENTER-...
* SAMPLE_ID must have less than 50 characters

### CODE REVIEW

**Comment by thomasyu888** [comment]
@hhunterzinck This can be reviewed, but do not merge yet.  We will wait until after the public release to merge

File: genie/process_functions.py
Comment (hhunterzinck) [inline genie/process_functions.py]:
```suggestion
    if any(identifiers.str.len() >= 50):
```

File: genie_registry/clinical.py
Comment (hhunterzinck) [inline genie_registry/clinical.py]:
```suggestion
            if any(clinicaldf[patientId].str.len() >= 50):
```

**Review by hhunterzinck — CHANGES_REQUESTED** [review]
@thomasyu888 I was trying to run with the SAGE test files but got tripped up since SAGE isn't a valid center in the reference table (syn10061452) from which the configuration loads the list of valid centers.  

`genie validate data_clinical_supp_SAGE.txt SAGE`

output the following error:

> Traceback (most recent call last):
>   File "/Users/hhunterzinck/opt/anaconda3/envs/genie/bin/genie", line 8, in <module>
>     sys.exit(main())
>   File "/Users/hhunterzinck/opt/anaconda3/envs/Genie/lib/python3.8/site-packages/genie/__main__.py", line 176, in main
>     args.func(syn, args)
>   File "/Users/hhunterzinck/opt/anaconda3/envs/Genie/lib/python3.8/site-packages/genie/validate.py", line 212, in _perform_validate
>     _check_center_input(args.center, list(genie_config["center_config"].keys()))
>   File "/Users/hhunterzinck/opt/anaconda3/envs/Genie/lib/python3.8/site-packages/genie/validate.py", line 175, in _check_center_input
>     raise ValueError(
> ValueError: Must specify one of these centers: JHU, DFCI, GRCC, NKI, MSK, UHN, VICC, MDA, WAKE, YALE, UCSF, CRUK, CHOP, VHIO, SCI, PROV, COLU, UCHI, DUKE

Could the SAGE site be included for testing without disrupting production processing of other sites?

**Comment by thomasyu888** [comment]
@hhunterzinck thanks for testing!  make sure to set the correct --project_id. It defaults to main genie project id, but if you provide the test genie project id, sage will be included

**Review by hhunterzinck — APPROVED** [review]
Great, thanks!  Was able to use the appropriate test project to test invalid sample and patient IDs.  Looks great.

**Comment by thomasyu888** [comment]
Thanks for the review. This is another PR that should be merged after the public release

**Comment by thomasyu888** [comment]
Use _validate_genie function for structural variants

**Pull request merged**
